up42/viztools.py

```diff
@@ -308,29 +322,42 @@ def map_results(
                 "You first need to download the results via job.download_results()!"
             )
 
-        # Add features to map.
-        # Some blocks store vector results in an additional geojson file.
-        # pylint: disable=not-an-iterable
-        json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
-        if json_fp:
-            json_fp = json_fp[0]
-        else:
+        f_paths = []
+        if isinstance(self.results, list):
+            # Add features to map.
+            # Some blocks store vector results in an additional geojson file.
             # pylint: disable=not-an-iterable
-            json_fp = [fp for fp in self.results if fp.endswith(".json")][0]
+            json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
+            if json_fp:
+                json_fp = json_fp[0]  # why only one element is selected?
+            else:
+                # pylint: disable=not-an-iterable
+                json_fp = [fp for fp in self.results if fp.endswith(".json")][0]
```

Comment:
I guess here it would make sense to pass the whole list instead of selecting just the first element? But your question to @chrieke will probably clarify

Reply:
Ideally, that's what I want to achieve. There has been no change in this part of the code from before. Hoping that, with some context it would bring more clarity!

Reply:
For each job result there is just a single json (and geojson), and we only want to visualize one of them. So getting the first element just gets rid of the list. Can also do that later in the code if it makes things easier.

Reply:
I guess in the case of the osm block, for instance, you can have multiple .geojson file each with different geometry.

Reply:
hm okay, that makes it more complicated, since this case is also not covered in the current code. Any more blocks that have multiple json/geojson files? If too complicated maybe create a separate ticket @marooned20, might require some further discussion.